src/deploy/main.py

```diff
@@ -1,15 +1,112 @@
+import json
 import os
+import sys
 
+import dill as dpickle
 import tensorflow as tf
 import wandb
 
-model = None
+
+class IssueLabeler:
+    def __init__(self, 
+                 body_text_preprocessor, 
+                 title_text_preprocessor, 
+                 model, 
+                 class_names=['bug', 'feature_request', 'question']):
+        """
+        Parameters
+        ----------
+        body_text_preprocessor: ktext.preprocess.processor
+            the text preprocessor trained on issue bodies
+        title_text_preprocessor: ktext.preprocess.processor
+            text preprocessor trained on issue titles
+        model: tensorflow.keras.models
+            a keras model that takes as input two tensors: vectorized 
+            issue body and issue title.
+        class_names: list
+            class names as they correspond to the integer indices supplied to the model. 
+        """
+        self.body_pp = body_text_preprocessor
+        self.title_pp = title_text_preprocessor
+        self.model = model
+        self.class_names = class_names
+        
+    
+    def get_probabilities(self, body:str, title:str):
+        """
+        Get probabilities for the each class. 
+        
+        Parameters
+        ----------
+        body: str
+           the issue body
+        title: str
+            the issue title
+            
+        Returns
+        ------
+        Dict[str:float]
+        
+        Example
+        -------
+        >>> issue_labeler = IssueLabeler(body_pp, title_pp, model)
+        >>> issue_labeler.get_probabilities('hello world', 'hello world')
+        {'bug': 0.08372017741203308,
+         'feature': 0.6401631832122803,
+         'question': 0.2761166989803314}
+        """
+        # transform raw text into array of ints
+        vec_body = self.body_pp.transform([body])
+        vec_title = self.title_pp.transform([title])
+        
+        # get predictions
+        probs = self.model.predict(x=[vec_body, vec_title]).tolist()[0]
+        
+        return {k:v for k, v in zip(self.class_names, probs)}
+
+# Build the Issue Labeler In The Global Scope
+#############################################
+global model
```

Comment:
I’m pretty sure you don’t need to declare global here

Reply:
yeah, it does not seem to make a difference, I will remove those.

Reply:
Another example is https://cloud.google.com/functions/docs/bestpractices/networking

It’s easy to misunderstand what they mean by global you don’t actually have to declare as global just means it’s defined as top level because according to their magic only the function scope is rerun.